src/gui/mzroll/ligandwidget.cpp

```diff
@@ -403,25 +404,38 @@ void LigandWidget::showTable() {
                 QTreeWidgetItem *item = new QTreeWidgetItem(parent, PathwayType);
                 item->setText(0,QString(compound->category[i].c_str()));
                 //parent->setData(0,Qt::UserRole,QVariant::fromValue(QString(pathway_id.c_str())));
-
             }*/
 
     }
-    treeWidget->setSortingEnabled(true);
+  
+    
+    CompoundsHash.clear();
+    QTreeWidgetItemIterator itr(treeWidget);
+        while(*itr)
+        {     
+        QTreeWidgetItem* item =(*itr);
+        QVariant v = item->data(0,Qt::UserRole);
+        Compound*  c =  v.value<Compound*>();                                                                                   
+        CompoundsHash.insert(c , item);
+        ++itr;
+        }
+treeWidget->setSortingEnabled(true);
 }
 
 void LigandWidget::markAsDone(Compound* compound) {
-
-    if (compound != NULL) {
-        QTreeWidgetItem* item = getItem(compound);
-
-        if (item != NULL) {
-            for (int col = 0; col < treeWidget->columnCount(); col++) {
-                item->setBackgroundColor(col, QColor(61, 204, 85, 100));
-            }
+    if(compound != NULL)
```

Comment:
@46navdeep I would suggest moving the code out of the 'if' block and just returning early.

Read more here: https://softwareengineering.stackexchange.com/questions/18454/should-i-return-from-a-function-early-or-use-an-if-statement